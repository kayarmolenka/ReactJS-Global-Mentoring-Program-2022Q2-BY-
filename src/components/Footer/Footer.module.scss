.footer {
    background-color: rgba(66, 66, 66, 1);
    height: 70px;
    display: flex;
    align-items: center;
    justify-content: center;
}