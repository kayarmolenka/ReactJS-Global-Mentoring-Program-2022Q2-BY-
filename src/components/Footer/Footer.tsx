import { NetflixRoulette } from '../NetflixRoulette';

import styles from './Footer.module.scss';

export const Footer = () => {
  return (
    <footer className={styles.footer}>
      <NetflixRoulette />
    </footer>
  );
};
