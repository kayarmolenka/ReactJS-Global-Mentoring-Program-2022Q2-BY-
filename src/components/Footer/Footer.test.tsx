import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Footer } from './Footer';

const HomePage = () => <div>{'HomePage'}</div>;
describe('Footer', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should return to Home page and show "HomePage" after click on logo in the footer', () => {
    render(
      <MemoryRouter initialEntries={['/footer']}>
        <Routes>
          <Route path="/" element={<HomePage />} />
          <Route path="/footer" element={<Footer />} />
        </Routes>
      </MemoryRouter>,
    );

    fireEvent.click(screen.getByText('netflix'));

    expect(screen.getByText('HomePage')).toBeInTheDocument();
  });
});
