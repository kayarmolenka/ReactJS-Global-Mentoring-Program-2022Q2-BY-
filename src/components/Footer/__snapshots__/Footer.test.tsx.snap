// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="logo"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <strong>
          netflix
        </strong>
        roulette
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
